Extension/src/LanguageServer/Providers/documentFormattingEditProvider.ts

```diff
@@ -16,6 +16,34 @@ export class DocumentFormattingEditProvider implements vscode.DocumentFormatting
     public async provideDocumentFormattingEdits(document: vscode.TextDocument, options: vscode.FormattingOptions, token: vscode.CancellationToken): Promise<vscode.TextEdit[]> {
         await this.client.awaitUntilLanguageClientReady();
         const filePath: string = document.uri.fsPath;
+        const onChanges: string | number | boolean = options.onChanges;
+        if (onChanges) {
+            let insertSpacesSet: boolean = false;
+            let tabSizeSet: boolean = false;
+            for (const editor of vscode.window.visibleTextEditors) {
+                if (document === editor.document) {
+                    if (editor.options.insertSpaces && typeof editor.options.insertSpaces === "boolean") {
+                        options.insertSpaces = editor.options.insertSpaces;
+                        insertSpacesSet = true;
+                    }
+                    if (editor.options.tabSize && typeof editor.options.tabSize === "number") {
+                        options.tabSize = editor.options.tabSize;
+                        tabSizeSet = true;
+                    }
+                    break;
+                }
+            };
+
+            if (!insertSpacesSet || !tabSizeSet) {
```

Comment:
Why do the two conditionals below need to be around the check "if (!insertSpacesSet || !tabSizeSet)" and then there are individual checks for each? It looks like unnecessary double checking.